.github/workflows/main.yml

```diff
@@ -1,42 +1,182 @@
-name: CI
+name: "lint & build changed ports"
 
 on:
   pull_request:
     branches:
-    - master
+      - master
     paths-ignore:
-    - '.github/**'
+      - '.github/**'
   push:
     branches:
-    - update/*
+      - update/*
 
 jobs:
   build:
+    name: ${{ matrix.os }}
     timeout-minutes: 0
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
         os: [macos-10.15, macos-11.0]
 
     steps:
-    - name: Checkout repository
-      uses: actions/checkout@v2
-      with:
-        fetch-depth: 64
-    - name: Bootstrap MacPorts
-      run: . _ci/bootstrap.sh
-    - name: Build
-      run: |
-        unset CC && source /opt/local/share/macports/setupenv.bash
-        export PATH="/opt/mports/bin:${PWD}/../mpbb:$PATH" && hash -r
-        sudo env PASTE_URL="$PASTE_URL" ./runner
-    - name: Fix log permission
-      if: ${{ always() }}
-      run: |
-        if [ -n "$(find /tmp/ -maxdepth 1 -name "ci-build-*" -print)" ]; then sudo mv /tmp/ci-build-* . && sudo chmod 755 ci-build-*; fi
-    - name: Archive build logs
-      if: ${{ always() }}
-      uses: actions/upload-artifact@v2
-      with:
-        name: port-build-log
-        path: ci-build-*/**/*.log
+      - name: Checkout ports
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 64
+          path: ports
+      - name: Checkout mpbb
+        uses: actions/checkout@v2
+        with:
+          repository: macports/mpbb
+          path: mpbb
+      - name: Bootstrap MacPorts
+        run: . ports/.github/workflows/bootstrap.sh
+      - name: Determine list of changed ports
+        id: portlist
+        run: |
+          set -eu
```

Comment:
Nice. Maybe we should make these bigger runs in scripts in `_ci/github` or even just `_ci`

Reply:
TBH I wanted to really get rid of _ci completely, there's .github that we have to create anyway, so I'd rather have it all there.

These steps should be centralized and, ideally, improved so that we don't have to maintain them twice for GitHub Actions and Azure Pipelines, but one thing at a time. I'd rather get this merged soon and iterate and improve later.

I think this is good to go as is, except I'd like to check whether GitHub Actions actually has a 10.14 image available. If so, I'd disable Azure Pipelines, since GitHub's integration is better. We can keep it around just in case we'll need it later, though.

Reply:
> I think this is good to go as is, except I'd like to check whether GitHub Actions actually has a 10.14 image available. If so, I'd disable Azure Pipelines, since GitHub's integration is better. We can keep it around just in case we'll need it later, though.

It does not. See https://github.com/macports/macports-ports/pull/8510.

Reply:
Thanks for checking.